api/api/constants/privilege.py

```diff
@@ -0,0 +1,72 @@
+import typing
+from dataclasses import dataclass
+
+from rest_framework.request import Request
+
+
+ANONYMOUS: typing.Literal["anonymous"] = "anonymous"
+AUTHENTICATED: typing.Literal["authenticated"] = "authenticated"
+PRIVILEGED: typing.Literal["privileged"] = "privileged"
+
+Level = typing.Literal[ANONYMOUS, AUTHENTICATED, PRIVILEGED]
+LEVELS = typing.get_args(Level)
+
+
+_PRIVILEGES = {}
+
+
+@dataclass
+class Privilege:
+    """
+    Privileges granted to applications upon approved request to Openverse maintainers.
+
+    Maintainers review requests for increased privileges on a per-case basis.
+
+    Distinct from ``rate_limit_model`` which only affects access rates rather than privileges.
+    """
+
+    slug: str
+    anonymous: typing.Any
```

Comment:
In practice, they're currently `int`, but I resume they could be anything. I'll try and see if Python will support a Generic parameter here :thinking:

Reply:
A generic appears to work! :tada: Well, whether it would actually pass mypy, I have no idea, but the code runs still and I don't get any errors with my informal type information from my IDE.